SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -237,12 +237,33 @@ ServerStatusWatcher {
 		}, AppClock)
 	}
 
-	prRecoverRemoteLogin { |clientID|
-		"% - recovering by adapting clientID %.\n".postf(server, clientID);
-		// make sure we can set the clientID
-		notified = false;
-		server.clientID_(clientID);
-		this.prFinalizeBoot;
+	// This method attempts to recover from a loss of client-server contact,
+	// which is a serious emergency in live shows. So it posts a lot of info
+	// on the recovered state, and possibly helpful next user actions.
+
+	prHandleLoginWhenAlreadyRegistered { |clientIDFromProcess|
+		"% - handling login request though already registered - \n".postf(server);
+
+		// by default, only reset clientID if changed, to leave allocators untouched:
+		if (clientIDFromProcess != server.clientID) {
+			// make sure we can set the clientID, and set it
+			notified = false;
+			server.clientID_(clientIDFromProcess);
+			"// This seems to be a login after a crash, or from a new server object,\n"
+			"// so you may want to release currently running synths by hand:\n".postln;
+			"%.defaultGroup.release;\n".postf(server.cs);
+			"// and you may want to redo server boot finalization by hand:\n".postln;
+			"%.statusWatcher.prFinalizeBoot;\n".postf(server.cs);
```

Comment:
If we want to suggest this line as something to actually do, it shouldn't involve a private message. What about adding:

```supercollider
+ Server {
		finalizeBoot { this.statusWatcher.prFinalizeBoot }
}
```

then the line above would read:
`"%.finalizeBoot;\n".postf(server.cs);`

Reply:
I'm not sure about the name ...

Reply:
IIUC this solution is a hack, so it's ok to call a private function. Adding access to a private function as a public function seems worse because it has to be walked back with deprecation.

Reply:
I see, makes some sense. So let's note it in the comment that this is a temporary measure which will be replaced in future versions?

My only worry is that the public use of private methods might be perceived as the standard way to go.

Reply:
like the comment above the method says, the whole thing is about making it possible to recover in an emergency, so yes, it is a hack.